src/main/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStep.java

```diff
@@ -126,7 +130,31 @@ public final void checkout(Run<?,?> run, FilePath workspace, TaskListener listen
                 }
                 }
             }
-            scm.checkout(run, launcher, workspace, listener, changelogFile, baseline);
+
+            for (int retryCount = Jenkins.get().getScmCheckoutRetryCount(); retryCount >= 0; retryCount--) {
+                try {
+                    scm.checkout(run, launcher, workspace, listener, changelogFile, baseline);
+                    break;
+                } catch (AbortException e) {
+                    // abort exception might have a null message.
+                    // If so, just skip echoing it.
+                    if (e.getMessage() != null) {
+                        listener.error(e.getMessage());
+                    }
+                } catch (InterruptedIOException e) {
+                    throw e;
+                } catch (Exception e) {
+                    // checkout error not yet reported
+                    Functions.printStackTrace(e, listener.error("Checkout failed"));
+                }
+
+                if (retryCount == 0)   // all attempts failed
+                    throw new AbortException("Maximum checkout retry attempts reached, aborting");
```

Comment:
Ideally we would throw up the original exception if this is the last attempt, rather than constructing a fresh exception.

Reply:
Fixed